src/btree/row_srch.c

```diff
@@ -614,14 +620,7 @@ leaf_match:	cbt->compare = 0;
 
 	return (0);
 
-err:	/*
-	 * Release the current page if the search started at the root. If the
-	 * search didn't start at the root we should never have gone looking
-	 * beyond the start page.
-	 */
-	WT_ASSERT(session, leaf == NULL || leaf == current);
-	if (leaf == NULL)
-		WT_TRET(__wt_page_release(session, current, 0));
+err:	WT_TRET(__wt_page_release(session, current, 0));
```

Comment:
@keithbostic I found this when testing out a custom collator. When the collator returned an error, I was ending up with a hazard pointer error (the page was being released twice). This change simplifies the logic around release - by ensuring that only one place references a page at once. The original change went in https://github.com/wiredtiger/wiredtiger/commit/f43b07032ce1cfa0e844a4365082b0e001e0a83d and in WT-2202